compiler/src/main/kotlin/io/github/apl_cornell/viaduct/codegeneration/BackendCodeGenerator.kt

```diff
@@ -234,9 +254,35 @@ private class BackendCodeGenerator(
             is AssertionNode -> TODO("Assertions not yet implemented.")
         }
     }
+    private fun argument(protocol: Protocol, argument: FunctionArgumentNode): CodeBlock {
+        return when (argument) {
+            // Input arguments
+            is ObjectReferenceArgumentNode -> {
+                when (typeAnalysis.type(nameAnalysis.declaration(argument))) {
```

Comment:
You don't need to case here. Functions calls of the form `f(&x)` where `x` is a `MutableCellType` are caught by the type checker. In fact, if they were allowed, not casing would be the correct thing to do.

Reply:
I'm not sure I understand. Do you mean that ObjRefArgNodes will always be VectorTypes, so I should just return the kotlinName of the argument?

Reply:
Every case should return
```CodeBlock.of("%N", context.kotlinName(argument.variable.value))```
irregardless of object type. So you should not case on `typeAnaylsis.type`.